### PR TITLE
Enhancement: Make name optional when running `flow.serve`

### PR DESCRIPTION
**Pull request by znicholasbrown**
### Example

This PR modifies the behavior of `flow.serve` to no longer require passing a name for the subsequently-created deployment, instead defaulting to the name of the flow. 

Example:
```py
@flow
def etl_flow():
    pass

if __name__ == "__main__":
  etl_flow.serve()
```

results in:
```
Your flow 'etl-flow' is being served and polling for scheduled runs!

To trigger a run for this flow, use the following command:

        $ prefect deployment run 'etl-flow/etl-flow'
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [x] This pull request includes helpful docstrings.
- [x] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: src/prefect/flows.py [hunk 1]
```diff
@@ -733,7 +733,7 @@ def my_other_flow(name):
     @sync_compatible
     async def serve(
         self,
-        name: str,
+        name: str = None,
```
Comment (zzstoatzz) [inline src/prefect/flows.py:736]: for the type checkers

```suggestion
        name: Optional[str] = None,
```

**Review by zzstoatzz — APPROVED** [review]
great call! thanks

File: src/prefect/flows.py (hunk 1 above)
Comment (serinamarie) [inline src/prefect/flows.py:736]:
```suggestion
        name: Optional[str] = None,
```

nit: I think we're trying to move toward being more explicit here. 

with: `from typing import Optional`

**Review by serinamarie — APPROVED** [review]
lgtm!

**Pull request merged**